DataBiosphere/terra-azure-relay-listeners
# [IA-4401] Allow get status endpoints without updating dateAccessed

### Jira Ticket: https://broadworkbench.atlassian.net/browse/IA-4401




## Summary of changes:


### What
- Filter requests which pass through SetDateAccessedInspector: if request includes the custom HTTP header `X-SetDateAccessedInspector-Action=ignore`, do not send a PATCH updateDateAccessed. This allows Leonardo to exempt its metrics and monitoring calls from this behavior, preventing a loop where Leo's checks for runtime health prevent the runtime from autopausing.

### Why
- Prevents Leo's MetricsMonitor from blocking Azure runtime autopause

### Testing strategy
- Create an Azure runtime with autopause >25m
- Verify it stops after that length of time idle
- Create apps (RStudio, Cromwell)
- Verify these function normally

- [x] I validated these changes on my BEE:
* autopause works with a DS runtime
* autopause works with an FS runtime
* autopause works with and without Cromwell running
* CRUD operations work on Azure runtimes
* autopause works on GCP runtimes (Jupyter, RStudio)
* CRUD operations work on GCP runtimes
